lib/index.js

```diff
@@ -140,6 +140,14 @@ function getOptions(options) {
   options.sourceMap = getSourceMap(options);
   options.style = getStyle(options) || 0;
 
+  // context object represents node-sass environment
+  options.context = {
+    info: module.exports.info,
+    state: {},
+    sync: false,
```

Comment:
@Jakobo, 

* node-sass is not complying with Ruby-Sass, since two runtimes are intrinsically different and we are more focused on node-way or JavaScript-way of doing things here.
* we are not restricting / filtering the options object anyway and the option object is open-ended. Users are free to consume it.
* info() have no or very obscure usage in importers.

> I do think there's a lot of value in people not writing directly into the top level hash or the option hash.

1- Why?
2- What is the purpose of having deep-nested clone?